osu.Game/Overlays/Dashboard/Friends/FriendsLayout.cs

```diff
@@ -0,0 +1,256 @@
+            userListToolbar.DisplayStyle.BindValueChanged(_ => recreatePanels());
+            userListToolbar.SortCriteria.BindValueChanged(_ => recreatePanels());
+
+            if (!api.IsLoggedIn)
+                return;
+
+            request = new GetFriendsRequest();
+            request.Success += response => Schedule(() => Users = response);
+            api.Queue(request);
+        }
+
+        private void recreatePanels()
+        {
+            // Don't allow any changes until we have users loaded
+            if (!usersLoaded)
+                return;
+
+            cancellationToken?.Cancel();
+
+            if (itemsPlaceholder.Any())
+                loading.Show();
+
+            var groupedUsers = onlineStatusControl.Current.Value?.Users ?? new List<APIFriend>();
+
+            var sortedUsers = sortUsers(groupedUsers);
+
+            LoadComponentAsync(createTable(sortedUsers), addContentToPlaceholder, (cancellationToken = new CancellationTokenSource()).Token);
+        }
+
+        private void addContentToPlaceholder(Drawable content)
+        {
+            loading.Hide();
+
+            var lastContent = currentContent;
+
+            if (lastContent != null)
+            {
+                lastContent.FadeOut(100, Easing.OutQuint).Expire();
+                lastContent.Delay(25).Schedule(() => lastContent.BypassAutoSizeAxes = Axes.Y);
+            }
+
+            itemsPlaceholder.Add(currentContent = content);
+            currentContent.FadeIn(200, Easing.OutQuint);
+        }
+
+        private FillFlowContainer createTable(List<APIFriend> users)
+        {
+            var style = userListToolbar.DisplayStyle.Value;
+
+            return new FillFlowContainer
+            {
+                RelativeSizeAxes = Axes.X,
+                AutoSizeAxes = Axes.Y,
+                Spacing = new Vector2(style == OverlayPanelDisplayStyle.Card ? 10 : 2),
+                Children = users.Select(u => createUserPanel(u, style)).ToList()
+            };
+        }
+
+        private UserPanel createUserPanel(User user, OverlayPanelDisplayStyle style)
+        {
+            switch (style)
+            {
+                default:
+                case OverlayPanelDisplayStyle.Card:
+                    return new UserGridPanel(user).With(panel =>
+                    {
+                        panel.Anchor = Anchor.TopCentre;
+                        panel.Origin = Anchor.TopCentre;
+                        panel.Width = 290;
+                    });
+
+                case OverlayPanelDisplayStyle.List:
+                    return new UserListPanel(user);
+            }
+        }
+
+        private List<APIFriend> sortUsers(List<APIFriend> unsorted)
+        {
+            switch (userListToolbar.SortCriteria.Value)
+            {
+                default:
+                case UserSortCriteria.LastVisit:
+                    return unsorted.OrderBy(u => u.LastVisit).Reverse().ToList();
```

Comment:
```suggestion
                    return unsorted.OrderByDescending(u => u.LastVisit).ToList();
```

Reply:
Before:
![my](https://user-images.githubusercontent.com/22874522/76805321-8dad2200-67ef-11ea-885d-8f2daa1568db.png)
After:
![Screenshot_1](https://user-images.githubusercontent.com/22874522/76805329-9140a900-67ef-11ea-9e62-e65ef26ee053.png)

Reply:
Not sure what your point is? The two last cards both have a `null` value on `LastVisit`. `OrderBy` is a stable sort (meaning that if two items have the same value of the attribute to sort by, it preserves the initial order of the items). The reason for the swap is, because you did the `Reverse` after sorting, the effects of the stable sort got reversed too. So the new behaviour is arguably the correct one.

Reply:
That's why we need an online test, which is currently in progress!

Reply:
Yeah, everything is fine